indexer/routing.hpp

```diff
@@ -0,0 +1,141 @@
+#pragma once
+
+#include "coding/bit_streams.hpp"
+#include "coding/elias_coder.hpp"
+#include "coding/reader.hpp"
+#include "coding/varint.hpp"
+#include "coding/write_to_sink.hpp"
+
+#include "base/assert.hpp"
+#include "base/bits.hpp"
+
+#include "std/vector.hpp"
+
+namespace routing
+{
+/// \brief Restriction to modify road graph.
+struct Restriction
+{
+  static uint32_t const kInvalidFeatureId;
+
+  /// \brief Types of road graph restrictions.
+  /// \note Despite the fact more that 10 restriction tags are present in osm all of them
+  /// could be split into two categories.
+  /// * no_left_turn, no_right_turn, no_u_turn and so on go to "No" category.
+  /// * only_left_turn, only_right_turn and so on go to "Only" category.
+  /// That's enough to rememeber if
+  /// * there's only way to pass the junction is driving along the restriction (Only)
+  /// * driving along the restriction is prohibited (No)
+  enum class Type
+  {
+    No,    // Going according such restriction is prohibited.
+    Only,  // Only going according such restriction is permitted
+  };
+
+  Restriction(Type type, size_t linkNumber);
```

Comment:
s/linkNumber/numLinks/

BTW, where this constructor is useful? I guess it's not a good idea to create a vector of invalid links. Why not to fill it directly with valid links?